tests/server/test_agents_api.py

```diff
@@ -216,6 +216,22 @@ async def test_agents_paging(server, client, env_with_agents: None, environment:
 
     assert result.result["metadata"] == {"total": 7, "before": 0, "after": 0, "page_size": 100}
 
+    fake_id_url = f"{base_url}/api/v2/agents?limit=1&sort=last_failover.asc&filter.status=paused&first_id=zzzz"
+    request = HTTPRequest(
+        url=fake_id_url,
+        headers={"X-Inmanta-tid": str(environment)},
+    )
+    response = await http_client.fetch(request, raise_error=False)
+    assert response.code == 200
+    response = json.loads(response.body.decode("utf-8"))
+    # We don't have a way to reconstruct the previous link
+    assert response["links"] == {
+        "first": "/api/v2/agents?limit=1&sort=last_failover.asc&filter.status=paused",
+        "prev": "/api/v2/agents?limit=1&sort=last_failover.asc&filter.status=paused&last_id=zzzz",
+    }
+    assert response["metadata"] == {"total": 2, "before": 2, "after": 0, "page_size": 1}
```

Comment:
What does `page_size` represent? Should it be 1?

Reply:
It's the number of objects that we asked for. I rewrote that part to make it clearer